hoymiles_mqtt/runners.py

```diff
@@ -33,19 +37,60 @@ def __init__(self, mqtt_builder: HassMqtt, mqtt_publisher: MqttPublisher, modbus
     def execute(self):
         """Get data from DTU and publish to MQTT broker."""
         is_acquired = self._lock.acquire(blocking=False)
-        if is_acquired:
-            if time.localtime().tm_hour == RESET_HOUR:
-                self._mqtt_builder.clear_production_today()
+        if not is_acquired:
+            return
+
+        if time.localtime().tm_hour == RESET_HOUR:
+            self._mqtt_builder.clear_production_today()
+            logger.info("Reset hour reached")
+
+        logger.debug("Read data from DTU")
+        plant_data = None
+        publish_count = 0
+        try:
+            plant_data = self._modbus_client.plant_data
+            logger.debug("Received data from DTU")
+        except pymodbus_exceptions.ModbusIOException as exc:
+            if 'No response received, expected at least 8 bytes' in exc.message:
+                logger.warning("Failed to read data from DTU via Modbus. Will retry.")
+            else:
+                logger.exception("Failed to read data from DTU via Modbus.")
+        except Exception:
+            logger.exception("Failed to read data from DTU. Unknown failure type.")
+
+        if plant_data:
             try:
-                plant_data = self._modbus_client.plant_data
+                # Publish configurations?
+                # This is done only for the first data set
                 if not self._mqtt_configured:
                     for topic, payload in self._mqtt_builder.get_configs(plant_data=plant_data):
                         self._mqtt_publisher.publish(topic=topic, message=payload, retain=True)
+                        mqtt_broker = "mqtt://{}:{}/{}".format(self._mqtt_publisher._mqtt_broker,
+                                                               self._mqtt_publisher._mqtt_port,
+                                                               topic)
+                        logger.debug("Published config into {}".format(mqtt_broker))
```

Comment:
Pre-formatted logging messages should be avoided. Otherwise, formatting is done even if the message is not logged.

Reply:
Improved code.